### PR TITLE
Use `List` rather than `Array`

### PR DESCRIPTION
**Pull request by garyb**
Resolves #29

Also updated the build/readme for consistency with the rest of our stuff.

### CODE REVIEW

File: src/Text/Parsing/Parser/Token.purs [hunk 1]
```diff
@@ -198,7 +202,7 @@ type GenTokenParser s m
         -- | trailing white space.
         symbol           :: String -> ParserT s m String,
         -- | `lexeme p` first applies parser `p` and than the `whiteSpace`
-        -- | parser, pureing the value of `p`. Every lexical
+        -- | parser, returning the value of `p`. Every lexical
```
Comment (paf31) [inline src/Text/Parsing/Parser/Token.purs]: Oops :)

Comment (garyb) [inline src/Text/Parsing/Parser/Token.purs]: I did it myself the other day :laughing:

Comment (garyb) [inline src/Text/Parsing/Parser/Token.purs]: puréeing

**Comment by paf31** [comment]
👍 Makes sense, thanks.

**Pull request merged**